File: backend/pkg/db/speaker.go
package db

type Speaker struct {
	ID        *int64  `json:"id" example:"1"`
	Email     *string `json:"email" example:"firstname.lastname@gmail.com"`
	FirstName *string `json:"firstName" example:"Bob"`
	LastName  *string `json:"lastName" example:"Smith"`
}

func NewSpeaker() Speaker {
	return Speaker{
		ID:        Int64Ptr(0),
		Email:     StringPtr(""),
		FirstName: StringPtr(""),
		LastName:  StringPtr(""),
	}
}

type SpeakerReaderWriterUpdaterDeleter interface {
	SpeakerReader
	SpeakerWriter
	SpeakerUpdater
	SpeakerDeleter
}

type SpeakerReader interface {
	ReadASpeaker(speakerID int64) (Speaker, error)
	ReadAllSpeakers() ([]Speaker, error)
}

type SpeakerWriter interface {
	WriteASpeaker(email *string, firstName *string, lastName *string) (int64, error)
}

type SpeakerUpdater interface {
	UpdateASpeaker(id int64, email *string, firstName *string, lastName *string) error
}

type SpeakerDeleter interface {
	DeleteASpeaker(id int64) error
}

File: frontend/src/app/data_models/speaker.ts
export class Speaker {
  id: number;
  firstName: string;
  lastName: string;
  email: string;

  constructor(firstName: string, lastName: string, email: string) {
    this.id = -1;
    this.firstName = firstName;
    this.lastName = lastName;
    this.email = email;
  }

  static getFullName(speaker: Speaker): string {
    var name = "";
    var needSpace = false;
    if ("firstName" in speaker && speaker.firstName !== null) {
      name += speaker.firstName;
      needSpace = true;
    }
    if ("lastName" in speaker && speaker.lastName !== null) {
      if (needSpace) {
        name += " ";
      }
      name += speaker.lastName;
    }
    return name;
  }
}
